flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -139,9 +143,35 @@ object UserDefinedFunctionUtils {
         s"Function class '${function.getClass.getCanonicalName}' does not implement at least " +
           s"one method named 'eval' which is public, not abstract and " +
           s"(in case of table functions) not static.")
-    } else {
-      methods
     }
+
+    verifyScalaVarargsAnnotation(methods)
+    methods
+  }
+
+  /**
+   * If users specified an @varargs, Scala will generate two methods indeed.
+   * If there does not exist corresponding varargs method of the Seq method,
+   * we will throw an ValidationException.
+   */
+  def verifyScalaVarargsAnnotation(methods: Array[Method]) = {
+    methods.foreach(method => {
+      val signatures = method.getParameterTypes
+      if (signatures.nonEmpty &&
+        signatures.last.getName.equals("scala.collection.Seq") &&
```

Comment:
I think this condition is too strict. It wouldn't allow `Seq`s in functions which is one of the most important classes in Scala. This does not work right now:

```
object Func16 extends ScalarFunction {

  def eval(a: Seq[String]): String = {
    a.mkString(", ")
  }
}
```